web/src/components/Icon/Icon.tsx

```diff
@@ -64,59 +47,43 @@ export const iconMap = {
 type IconType = keyof typeof iconMap;
 type IconSizeType = "xs" | "sm" | "md" | "lg" | number;
 
-export const Icon = ({ icon, color, size, rotate, ...rest }: Props) => {
-  const IconSvg = iconMap[icon];
-  if (!IconSvg) return null;
-  return (
-    <StyledIcon
-      className="icon"
-      iconColor={color}
-      iconSize={size}
-      rotate={rotate}
-      {...rest}
-    >
-      <IconSvg />
-    </StyledIcon>
-  );
-};
-
-type StyledProps = {
-  iconColor?: keyof typeof colors;
-  iconSize?: IconSizeType;
-  rotate?: string;
-};
-
 const iconSizeMap = {
-  xs: "12px",
-  sm: "16px",
-  md: "24px",
-  lg: "40px",
+  xs: "w-3 h-3",
+  sm: "w-4 h-4",
+  md: "w-6 h-6",
+  lg: "w-10 h-10",
 };
-
 const getDimensions = (iconSize: IconSizeType) => {
-  if (typeof iconSize === "number") return `${iconSize}px`;
+  if (typeof iconSize === "number") return `w-[${iconSize}px] h-[${iconSize}px]`;
   return iconSizeMap[iconSize];
 };
 
-const getSvgStyles = (props: StyledProps) => {
-  const { iconSize, iconColor, rotate } = props;
+const StyledIcon = classed.div("inline-flex p-1 border-2 border-grey-dark dark:border-grey-light rounded-full");
 
-  return css`
-    width: ${iconSize ? getDimensions(iconSize) : "initial"};
-    height: ${iconSize ? getDimensions(iconSize) : "initial"};
-    transform: rotate(${rotate || 0});
-    transition: fill 0.23s ease;
-    path {
-      fill: ${iconColor ? theme.colors[iconColor] : "currentColor"};
-    }
-  `;
+type Props = ComponentProps<typeof StyledIcon> &
+{
+  icon: IconType;
+  // The color is expected to be a tailwind color
+  color?: string;
+  size?: IconSizeType;
+  rotate?: string;
 };
 
-const StyledIcon = styled.div<StyledProps>`
-  display: inline-flex;
-  padding: ${theme.spacings(1)};
+export const Icon = deriveClassed<typeof StyledIcon, Props>(({ icon, color, size, rotate, ...rest }, ref) => {
+  const IconSvg = iconMap[icon];
+  if (!IconSvg) return null;
+
+  const StyledSvg = classed(
+    IconSvg,
+    `rotate-[${rotate || 0}deg]`,
+    `transition-[fill] duration-[0.23s] ease-[ease]`,
+    `${color ? "svg-path:fill-" + color : "svg-path:fill-black dark:svg-path:fill-white"}`,
```

Comment:
same with this, look at above comment

Reply:
I'll rewrite the Icon component and is it alright to use the css variable trick?

Reply:
I rewrote the component, It now has variants for all the variables and if it's going to be a custom value it can be assigned by first selecting the `class` key of a variant ; for example:
`
<Icon icon="close" fillColor="class" className="svg-path:fill-red" />
`